src/Dropdown.php

```diff
@@ -1613,11 +1611,58 @@ public static function showItemType($types = '', $options = [])
                 'display'             => $params['display'],
                 'rand'                => $params['rand'],
                 'track_changes'       => $params['track_changes'],
+                'init'                => $params['init'],
+                'width'               => $params['width'],
+                'aria_label'          => $params['aria_label'],
             ]);
         }
         return 0;
     }
 
+    /**
+     * Build dropdown options and check rights if needed
+     *
+     * This method dynamically builds dropdown options based on the provided types.
+     * It supports both single and multiple types (nested) and checks access rights if required.
+     *
+     * @param array|string $types The types for which to build dropdown options. Can be a single type or an array of types.
+     * @param bool         $checkright Whether to check access rights for the type(s).
+     * @return array|string|null The built dropdown options, or null if access is denied or type is invalid.
+     */
+    public static function buildDropdownOptions($types, $checkright = false): array|string|null
+    {
+        $options = []; // Initialize the options array to accumulate results.
+
+        // Handle the case where $types is not an array, implying a single type.
+        if (!is_array($types)) {
+            if ($item = getItemForItemtype($types)) {
+                if ($checkright && !$item->canView()) {
+                    return null; // Access denied, return null.
+                }
+                // Access granted, return the type name.
+                return $item->getTypeName();
+            }
+
+            return null; // Item not found or invalid type, return null.
+        }
```

Comment:
It makes the signature pretty complex. Why not accepting only an array for the `$types` param, and handle this directly in the foreach loop ?

```php
foreach ($types as $label => $type) {
    if (!is_array($type)) {
        if (
            $item = getItemForItemtype($type)
            && (!$checkright || $item->canView())
        ) {
            $options[$type] = $item->getTypeName();
        }
        continue;
    }

    // Recursively build dropdown options for the current type.
    $opts = self::buildDropdownOptions($type, $checkright);
    if ($opts !== null) {
        $options[$label] = $opts;
    }
}
```